### PR TITLE
⚠️SecurityGroupFilter to SecurityGroupParam

### PR DESCRIPTION
**Pull request by mdbooth**
See https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1975

TODO:
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1972
- [x] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1971
- [x] [Docs](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1982)

/hold

### CODE REVIEW

File: api/v1alpha6/types_conversion.go [hunk 1]
```diff
@@ -41,46 +41,82 @@ func restorev1alpha6SecurityGroupFilter(previous *SecurityGroupFilter, dst *Secu
...
+		if err != nil {
+			return err
+		}
 	}
+
 	return nil
 }
 
-func Convert_v1alpha6_SecurityGroupFilter_To_v1beta1_SecurityGroupFilter(in *SecurityGroupFilter, out *infrav1.SecurityGroupFilter, s apiconversion.Scope) error {
+/*
```
Comment (dulek) [inline api/v1alpha6/types_conversion.go:103]: This looks weird?

File: api/v1beta1/types.go [hunk 2]
```diff
@@ -88,15 +88,40 @@ func (f *FilterByNeutronTags) IsZero() bool {
...
+// SecurityGroupFilter specifies a query to select an OpenStack security group. At least one property must be set.
+// +kubebuilder:validation:MinProperties:=1
 type SecurityGroupFilter struct {
-	ID          string `json:"id,omitempty"`
 	Name        string `json:"name,omitempty"`
 	Description string `json:"description,omitempty"`
 	ProjectID   string `json:"projectID,omitempty"`
 
 	FilterByNeutronTags `json:",inline"`
 }
```
Comment (dulek) [inline api/v1beta1/types.go:113]: We could probably take the opportunity to add descriptions here.

**Review by dulek — CHANGES_REQUESTED** [review]

File: api/v1alpha6/types_conversion.go (hunk 1 above)
Comment (mdbooth) [reply api/v1alpha6/types_conversion.go:103]: Eurgh. Forgot to remove this :(

**Comment by mdbooth** [comment]
Last push simply removes the commented code.

File: api/v1beta1/types.go (hunk 2 above)
Comment (mdbooth) [reply api/v1beta1/types.go:113]: I agree, but I'm trying to minimise this series as much as possible. These can probably all do with pointerification, too, but both can happen post v1beta1 release so I'd prefer to punt.

**Comment by mdbooth** [comment]
/hold cancel

I'd re-run the full test, but as I literally only removed some commented-out-code since it last ran successfully on this PR I won't.
(- Words on my tombstone)

**Pull request merged**